src/main/java/com/monumental/util/search/SearchHelper.java

```diff
@@ -0,0 +1,168 @@
+package com.monumental.util.search;
+
+import javax.persistence.criteria.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.monumental.util.string.StringHelper.isNullOrEmpty;
+
+/**
+ * Class containing static methods useful for building JPA Criteria API queries
+ */
+public class SearchHelper {
```

Comment:
I decided to put a lot of the different searching logic we were repeating into this class so we can re-use them easier.

Reply:
Very nice